internal/tools/neo4j/neo4jschema/neo4jschema.go

```diff
@@ -0,0 +1,712 @@
+// This function is called by the tool registry to create a new configuration object.
+func newConfig(ctx context.Context, name string, decoder *yaml.Decoder) (tools.ToolConfig, error) {
+	actual := Config{Name: name}
+	if err := decoder.DecodeContext(ctx, &actual); err != nil {
+		return nil, err
+	}
+	return actual, nil
+}
+
+// compatibleSource defines the interface a data source must implement to be used by this tool.
+// It ensures that the source can provide a Neo4j driver and database name.
+type compatibleSource interface {
+	Neo4jDriver() neo4j.DriverWithContext
+	Neo4jDatabase() string
+}
+
+// Statically verify that our compatible source implementation is valid.
+var _ compatibleSource = &neo4jsc.Source{}
+
+// compatibleSources lists the kinds of sources that are compatible with this tool.
+var compatibleSources = [...]string{neo4jsc.SourceKind}
+
+// Config holds the configuration settings for the Neo4j schema tool.
+// These settings are typically read from a YAML file.
+type Config struct {
+	Name               string   `yaml:"name" validate:"required"`
+	Kind               string   `yaml:"kind" validate:"required"`
+	Source             string   `yaml:"source" validate:"required"`
+	Description        string   `yaml:"description" validate:"required"`
+	AuthRequired       []string `yaml:"authRequired"`
+	CacheExpireMinutes *int     `yaml:"cacheExpireMinutes,omitempty"` // Cache expiration time in minutes.
+}
+
+// Statically verify that Config implements the tools.ToolConfig interface.
+var _ tools.ToolConfig = Config{}
+
+// ToolConfigKind returns the kind of this tool configuration.
+func (cfg Config) ToolConfigKind() string {
+	return kind
+}
+
+// Initialize sets up the tool with its dependencies and returns a ready-to-use Tool instance.
+func (cfg Config) Initialize(srcs map[string]sources.Source) (tools.Tool, error) {
+	// Verify that the specified source exists.
+	rawS, ok := srcs[cfg.Source]
+	if !ok {
+		return nil, fmt.Errorf("no source named %q configured", cfg.Source)
+	}
+
+	// Verify the source is of a compatible kind.
+	s, ok := rawS.(compatibleSource)
+	if !ok {
+		return nil, fmt.Errorf("invalid source for %q tool: source kind must be one of %q", kind, compatibleSources)
+	}
+
+	parameters := tools.Parameters{}
+	mcpManifest := tools.McpManifest{
+		Name:        cfg.Name,
+		Description: cfg.Description,
+		InputSchema: parameters.McpManifest(),
+	}
+
+	// Set a default cache expiration if not provided in the configuration.
+	if cfg.CacheExpireMinutes == nil {
+		defaultExpiration := cache.DefaultExpiration // Default to 60 minutes
+		cfg.CacheExpireMinutes = &defaultExpiration
+	}
+
+	// Finish tool setup by creating the Tool instance.
+	t := Tool{
+		Name:         cfg.Name,
+		Kind:         kind,
+		AuthRequired: cfg.AuthRequired,
+		Driver:       s.Neo4jDriver(),
+		Database:     s.Neo4jDatabase(),
+		cache:        cache.NewCache(),
+		conf:         &cfg,
```

Comment:
```suggestion
		cacheExpireMinutes: cfg.CacheExpireMinutes,
```

Seems like cacheExpireMinutes is the only thing that we need.